# Call Center Cleanup

For this assignment, we will be working with call center data. You can start working on the assignment after the first lesson on Exploratory Data Analysis. Make sure to read the whole assignment before starting anything! As you code along in the Jupyter notebook, you are asked to make note of the results of your analysis. Do so by clicking on the results box and adding your notes beside each question.

## Business Issue and Understanding

You are working for a company that has two call centers: the North Call Center and the South Call Center. The company is looking to possibly hire five additional reps to enhance customer experience. Your task is to explore how efficient the current reps are in each branch to determine which branch would benefit from additional assistance.

### How the Call Center Works

Call center representatives are assigned queues. When calls are assigned to a queue, the call is assigned to the next person in line in the queue. After a call is assigned to a representative, the amount of time between assignment and the call starting is divided into busy minutes and not ready minutes. If the call is incoming and a customer is waiting on the phone for a rep, the time is split into three categories: busy minutes, not ready minutes, and incoming call wait time. Once the rep has the customer on the phone, there might be during call wait time, where the call is put on hold while the rep gets an answer for the customer.

### Notes about the Dataset

If you haven't worked in a call center before, these notes might help you throughout your analysis.

* The call purpose is tagged for each call.
* The time of the call is tagged in 1 hour blocks starting at 9:00 AM and ending at 5:00 PM.
* Calls are tagged as incoming or outgoing.
* Reps are assigned to queues. When the reps are working, they take calls in the order of their queue.
* A call that is dropped due to technical error or missed by the center because they have reached maximum capacity is a lost call.
* An abandoned call is when the customer hangs up because they have been waiting for too long.
* Busy Minutes: the amount of time after a call comes in or needs to go out where the assigned rep is not available because they are busy with other customers.
* Not Ready Minutes: the amount of time after a call comes in or needs to go out where the assigned rep is not available because they are not ready (for example, getting water).
* Incoming Wait Time - amount of time after assigned rep is available to take the call customer waits for representative to pick up a call. This is tracked in seconds.
* During Call Wait Time - amount of time during call that customer has to wait for representative

## Getting Started

You have two CSVs at your disposal, `NorthCallCenter.csv` and `SouthCallCenter.csv`. Import the appropriate libraries and create two dataframes, one called `north_df` and one called `south_df`.

In [2]:
# Import the appropriate libraries with aliases
import pandas as pd
import numpy as np
import seaborn as sb
import matplotlib.pyplot as plt
from matplotlib import style
# setting style for graphs
style.use('ggplot')
plt.rcParams['figure.figsize'] = (20,10)
# Create two new dataframes
north_df = pd.read_csv('NorthCallCenter.csv')
south_df = pd.read_csv('SouthCallCenter.csv')
# north_df.head(5)
south_df.tail(50)

Unnamed: 0 Branch     Call Purpose Time Block Incoming or Outgoing Queue  \
264         264  South  Product Support    2:00 PM             Outgoing     C   
265         265  South  Product Support    2:00 PM             Outgoing     D   
266         266  South  Product Support    2:00 PM             Outgoing     D   
267         267  South  Product Support    2:00 PM             Outgoing     D   
268         268  South  Product Support    2:00 PM             Outgoing     C   
269         269  South  Product Support    2:00 PM             Outgoing     C   
270         270  South  Product Support    2:00 PM             Outgoing     C   
271         271  South  Product Support    3:00 PM             Outgoing     C   
272         272  South  Product Support    3:00 PM             Outgoing     C   
273         273  South  Product Support    3:00 PM             Outgoing     D   
274         274  South  Product Support    3:00 PM             Outgoing     D   
275         275  South  Product Support    3:00 PM             Outgoing     C   
276         276  South  Product Support    3:00 PM             Outgoing     D   
277         277  South  Product Support    3:00 PM             Outgoing     C   
278         278  South  Product Support    3:00 PM             Outgoing     C   
279         279  South  Product Support    3:00 PM             Outgoing     D   
280         280  South  Product Support    3:00 PM             Outgoing     C   
281         281  South  Product Support    3:00 PM             Outgoing     D   
282         282  South  Product Support    3:00 PM             Outgoing     D   
283         283  South  Product Support    3:00 PM             Outgoing     D   
284         284  South  Product Support    3:00 PM             Outgoing     C   
285         285  South  Product Support    3:00 PM             Outgoing     C   
286         286  South  Product Support    3:00 PM             Outgoing     D   
287         287  South  Product Support    3:00 PM             Outgoing     D   
288         288  South  Product Support    3:00 PM             Outgoing     D   
289         289  South  Product Support    3:00 PM             Outgoing     C   
290         290  South  Product Support    3:00 PM             Outgoing     D   
291         291  South  Product Support    4:00 PM             Outgoing     D   
292         292  South  Product Support    4:00 PM             Outgoing     C   
293         293  South  Product Support    4:00 PM             Outgoing     C   
294         294  South  Product Support    4:00 PM             Outgoing     D   
295         295  South  Product Support    4:00 PM             Outgoing     C   
296         296  South  Product Support    4:00 PM             Outgoing     C   
297         297  South  Product Support    4:00 PM             Outgoing     D   
298         298  South  Product Support    4:00 PM             Outgoing     D   
299         299  South  Product Support    4:00 PM             Outgoing     C   
300         300  South  Product Support    4:00 PM             Outgoing     D   
301         301  South  Product Support    4:00 PM             Outgoing     D   
302         302  South  Product Support    5:00 PM             Outgoing     C   
303         303  South  Product Support    5:00 PM             Outgoing     C   
304         304  South  Product Support    5:00 PM             Outgoing     D   
305         305  South  Product Support    5:00 PM             Outgoing     C   
306         306  South  Product Support    5:00 PM             Outgoing     D   
307         307  South  Product Support    5:00 PM             Outgoing     D   
308         308  South  Product Support    5:00 PM             Outgoing     C   
309         309  South  Product Support    5:00 PM             Outgoing     D   
310         310  South  Product Support    5:00 PM             Outgoing     C   
311         311  South  Product Support    5:00 PM             Outgoing     C   
312         312  South  Product Supp

<span style="background-color:orangered; color:orangered;">- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -</span>

## Exploratory Data Analysis

Time to do some EDA! In the process of learning more about the two datasets, answer the following questions. In the code blocks below enter the code used to solve the question.  At the end of the section record your answers by double clicking where asked.

#### EDA Question 1A:  How many reps are in the North branch?  245

In [3]:
#Code Used:
north_df.info("Time Block")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 245 entries, 0 to 244
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Unnamed: 0             245 non-null    int64  
 1   Branch                 245 non-null    object 
 2   Call Purpose           245 non-null    object 
 3   Time Block             245 non-null    object 
 4   Incoming or Outgoing   245 non-null    object 
 5   Queue                  245 non-null    object 
 6   Rep ID                 245 non-null    object 
 7   Sale                   245 non-null    object 
 8   Lost Call              245 non-null    int64  
 9   Abandoned              245 non-null    int64  
 10  Busy Minutes           245 non-null    int64  
 11  Not Ready Minutes      245 non-null    int64  
 12  Incoming Wait Time     82 non-null     float64
 13  During Call Wait Time  245 non-null    int64  
 14  Calls                  245 non-null    int64  
dtypes: flo

#### EDA Question 1B:  How many reps are in the South branch?  314

In [4]:
#Code Used:
south_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 314 entries, 0 to 313
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Unnamed: 0             314 non-null    int64  
 1   Branch                 314 non-null    object 
 2   Call Purpose           314 non-null    object 
 3   Time Block             314 non-null    object 
 4   Incoming or Outgoing   314 non-null    object 
 5   Queue                  314 non-null    object 
 6   Rep ID                 314 non-null    object 
 7   Sale                   314 non-null    object 
 8   Lost Call              314 non-null    int64  
 9   Abandoned              314 non-null    int64  
 10  Busy Minutes           314 non-null    int64  
 11  Not Ready Minutes      314 non-null    int64  
 12  Incoming Wait Time     126 non-null    float64
 13  During Call Wait Time  314 non-null    int64  
 14  Calls                  314 non-null    int64  
dtypes: flo

#### EDA Question 2A:  What is the average busy minutes, not ready minutes, incoming wait time, and during call wait time for the North branch? 

In [5]:
#Code Used:
busy_north = north_df['Busy Minutes']
busy_north_mean = busy_north.mean()
busy_north_mean

nr_north = north_df['Not Ready Minutes']
nr_north_mean = nr_north.mean()
nr_north_mean

in_north = north_df['Incoming Wait Time']
in_north_mean = in_north.mean()
in_north_mean

during_north = north_df['During Call Wait Time']
during_north_mean = during_north.mean()
during_north_mean

print( "Avg. North Busy Minutes: " + str(busy_north_mean))
print( "Avg. North Not Ready Minutes: " + str(nr_north_mean))
print( "Avg. North Incoming Wait Time: " + str(in_north_mean))
print( "Avg. North During Call Wait Time: " + str(during_north_mean))

Avg. North Busy Minutes: 9.987755102040817
Avg. North Not Ready Minutes: 1.9142857142857144
Avg. North Incoming Wait Time: 3.048780487804878
Avg. North During Call Wait Time: 2.9673469387755103


#### EDA Question 2B:  What is the average busy minutes, not ready minutes, incoming wait time, and during call wait time for the South branch? 

In [6]:
#Code Used:
busy_south = south_df['Busy Minutes']
busy_south_mean = busy_south.mean()
busy_south_mean

nr_south = south_df['Not Ready Minutes']
nr_south_mean = nr_south.mean()
nr_south_mean

in_south = south_df['Incoming Wait Time']
in_south_mean = in_south.mean()
in_south_mean

during_south = south_df['During Call Wait Time']
during_south_mean = during_south.mean()
during_south_mean

print( "Avg. South Busy Minutes: " + str(busy_south_mean))
print( "Avg. South Not Ready Minutes: " + str(nr_south_mean))
print( "Avg. South Incoming Wait Time: " + str(in_south_mean))
print( "Avg. South During Call Wait Time: " + str(during_south_mean))

Avg. South Busy Minutes: 10.054140127388536
Avg. South Not Ready Minutes: 1.910828025477707
Avg. South Incoming Wait Time: 3.0
Avg. South During Call Wait Time: 3.0764331210191083


#### EDA Question 3:  What is the number of calls taken for each time block(Both North and South combined)?

In [7]:
# Code Used:
# movies_df['genre'].describe()
# north_df['Time Block'].value_counts().head(10)

tb_north = north_df['Time Block']
print(tb_north.value_counts().head(10))
tb_south = south_df['Time Block']
print(tb_south.value_counts().head(10))

12:00 PM    45
10:00 AM    43
2:00 PM     35
3:00 PM     30
4:00 PM     28
11:00 AM    25
1:00 PM     17
5:00 PM     17
9:00 AM      5
Name: Time Block, dtype: int64
12:00 PM    75
10:00 AM    56
3:00 PM     43
11:00 AM    31
2:00 PM     30
5:00 PM     26
4:00 PM     25
1:00 PM     23
9:00 AM      5
Name: Time Block, dtype: int64


##### Record your answers for the EDA section by double clicking here:
- 1a) Reps in the North Branch = 245
- 1b) Reps in the South Branch = 314


- 2a) North Branch, (round to the nearest hundredth):
> - Busy Minutes = 9.987755102040817
> - Not Ready Minutes = 1.9142857142857144
> - Incoming Wait Time = 3.048780487804878
> - During Call Wait Time = 2.9673469387755103
- 2b) South Branch, (round to the nearest hundredth):
> - Busy Minutes = 10.054140127388536
> - Not Ready Minutes = 1.910828025477707
> - Incoming Wait Time = 3.0
> - During Call Wait Time = 3.0764331210191083
- 3) Total Calls taken for all branches per time block:
> - 9:00AM   = 10
> - 10:00AM  = 99
> - 11:00AM  = 56
> - 12:00PM  = 120
> - 1:00PM   = 40
> - 2:00PM   = 65
> - 3:00PM   = 73
> - 4:00PM   = 53
> - 5:00PM   = 43



<span style="background-color:orangered; color:orangered;">- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -</span>

## Cleaning Your Data 
Now you need to clean up the datasets. When cleaning the datasets, you may find that there isn't dirty data to clean. That is okay! Some questions you need to answer about the data sets.
* Add additional code blocks as needed to show how you came to your conclusions. Add comments in your code blocks to help others understand your thinking. 

#### Cleaning Question 1:  Is there any missing data and if yes explain what you would do with the missing data and why?
##### Record your answer by double clicking here:
> Your Answer: Yes, the Incoming Wait Time variable is missing quite a few times from both call centers' data. I'm going to replace the 'NaN' data entries with '0.0' because I still want to compare the 'Nan' entries with the non null entries.

In [8]:
# Code Used
# Add Additional Cells as needed.
# north_df.isnull().sum()
north_df.fillna(0.0, inplace = True)
north_df.tail(5)
# south_df.fillna(0.0, inplace = True)
# south_df.tail(5)
# south_df.isnull().sum()

Unnamed: 0 Branch     Call Purpose Time Block Incoming or Outgoing Queue  \
240         240  North  Product Support    5:00 PM             Outgoing     B   
241         241  North  Product Support    5:00 PM             Outgoing     A   
242         242  North  Product Support    5:00 PM             Outgoing     A   
243         243  North  Product Support    5:00 PM             Outgoing     A   
244         244  North  Product Support    5:00 PM             Outgoing     B   

     Rep ID Sale  Lost Call  Abandoned  Busy Minutes  Not Ready Minutes  \
240   Brent   NO          0          0            10                  2   
241     Joe   NO          0          0             8                  3   
242  Amanda   NO          0          0            12                  1   
243     Cam   NO          0          0            11                  2   
244  Xander   NO          0          0             8                  1   

     Incoming Wait Time  During Call Wait Time  Calls  
240                 0.0                      3      1  
241                 0.0                      2      1  
242                 0.0                      2      1  
243                 0.0                      2      1  
244                 0.0                      4      1

#### Cleaning Question 2:  In the North data set, there are two separate "YES" values for a sale. Why and how did you rectify the issue?
##### Record your answer by double clicking here:
> Your Answer: If I didnt rectify the wrong format there would be a third category reflected. I first checked to see what the value counts were and realized the lone 'YES' had an extra space in it. I then stripped that spaceto make for the two logical values, "YES" and "NO".

In [9]:
#Code Used
# Add Additional Cells as needed.
# north_df['Sale'].value_counts()
# north_df['Sale'] = north_df['Sale'].str.lower()
north_df['Sale'] = north_df['Sale'].str.strip() 
north_df['Sale'].value_counts()


NO     234
YES     11
Name: Sale, dtype: int64

#### Cleaning Question 3:  Are there duplicates in the two data sets? If there are, how did you handle them?
##### Record your answer by double clicking here:
> Your Answer: No, I did not find any duplicates. 

In [10]:
#Code Used
# Add Additional Cells as needed.
# north_df.duplicated().value_counts()
south_df.duplicated().value_counts()

False    314
dtype: int64

#### Cleaning Question 4:  Is any of the data in the two data sets unnecessary? If yes, how did you handle it?
##### Record your answer by double clicking here:
> Your Answer: Unnamed, Branch, Call Purpose and Calls are not needed. I dropped them.

In [11]:
#Code Used
# Add Additional Cells as needed.
# print(north_df.info())

north_df.drop(['Unnamed: 0', 'Call Purpose'], axis=1, inplace=True)
south_df.drop(['Unnamed: 0', 'Call Purpose'], axis=1, inplace=True)
# print(north_df.columns)
# print(south_df.columns)
# south_df.head()

<span style="background-color:orangered; color:orangered;">- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -</span>

## Data Manipulation
Before you begin answering any questions, combine the two datasets together to create a third dataframe called df. You can use this third dataframe to compare the two call centers to company-wide trends.
* Add additional code blocks as needed to show how you came to your conclusions. Add comments in your code blocks to help others understand your thinking. Record your answer by double clicking the below cell.

In [12]:
# Create dataframe for the entire company named df

df = pd.merge(north_df, south_df,how = 'outer')
df.tail(6)

Branch Time Block Incoming or Outgoing Queue  Rep ID Sale  Lost Call  \
553  South    5:00 PM             Outgoing     C    Eric   NO          0   
554  South    5:00 PM             Outgoing     D   Helga   NO          0   
555  South    5:00 PM             Outgoing     C   Susan   NO          0   
556  South    5:00 PM             Outgoing     C   Sandy   NO          0   
557  South    5:00 PM             Outgoing     C    Eric   NO          0   
558  South    5:00 PM             Outgoing     D  George   NO          0   

     Abandoned  Busy Minutes  Not Ready Minutes  Incoming Wait Time  \
553          0            11                  1                 NaN   
554          0            10                  3                 NaN   
555          0            12                  3                 NaN   
556          0             9                  3                 NaN   
557          0            10                  1                 NaN   
558          0            10                  2                 NaN   

     During Call Wait Time  Calls  
553                      4      1  
554                      4      1  
555                      4      1  
556                      4      1  
557                      2      1  
558                      3      1

#### Manipulation Question 1:  Group by Rep ID and sum the resulting structure. Sort by calls to determine which rep in each branch has the highest number of calls.

##### Record your answer by double clicking here:
Rep with the hightest number of calls and their total calls:
- North Branch = Brent has 37 calls
- South Branch = Eric has 35 calls


In [13]:
# Code Used
# Add Additional Cells as needed. 
# print(df.columns)
rep = df.groupby(['Rep ID','Branch'])
print(rep.sum())
# print(rep.aggregate({"Calls":['sum']}))
# rep.index.value_counts()

               Lost Call  Abandoned  Busy Minutes  Not Ready Minutes  \
Rep ID Branch                                                          
Alice  South           0          0           199                 46   
Amanda North           0          1           229                 43   
Andy   North           0          3           228                 43   
Brent  North           0          1           362                 71   
Cam    North           1          0           233                 46   
Duke   North           2          0           292                 56   
Eric   South           0          0           361                 54   
George South           0          0           298                 53   
Helga  South           0          2           292                 58   
Joe    North           0          0           256                 49   
Josh   South           3          0           260                 50   
Karl   South           1          0           287               

In [14]:
# calls = df.column('Calls')
# print(calls.sum())
# df.aggregate({"Calls":['sum', 'min']})
# south_df.aggregate({"Calls":['sum', 'min']})
north_df.aggregate({"Calls":['sum']})


Calls
sum    245

#### Manipulation Question 2:  The average call abandonment rate is a KPI when it comes to determining call center efficiency. As you may recall, abandoned calls are calls where the customer hangs up due to long call times. What is the average call abandonment rate for each branch and the whole company? Do any of these fall out of the optimal range of 2-5%?
##### Record your answer by double clicking here:
Average Call Abandonment Rates (round to the nearest hundredth):
- North Branch = 2.6% of 245 calls total
- South Branch = 1.27% of 314 calls total
- Company Wide = 1.97% of 559 calls total
- Do any of these fall out of the optimal range of 2-5%? No

In [15]:
# Code Used
# Add Additional Cells as needed.
#  245/314/559
#  7/4/11
# south_df.aggregate({'Abandoned' : ['sum']})
north_df.aggregate({'Abandoned' : ['sum']})
df.aggregate({'Abandoned' : ['sum']})

Abandoned
sum         11

#### Manipulation Question 3:  Service level is another KPI when it comes to measuring call center efficiency. Service level is the percentage of calls answered within a specific number of seconds. In the case of your employer, their ideal time frame is 2 seconds. What is the percentage of calls answered within 2 seconds for each branch and the entire company?
##### Record your answer by double clicking here:
Percentage of calls answered within 2 seconds, include # of calls:
- North Branch = 
- South Branch = 
- Company Wide = 

In [16]:
# Code Used
# Add Additional Cells as needed.
rep = df.groupby(['Rep ID','Branch'])
print(rep.sum())

               Lost Call  Abandoned  Busy Minutes  Not Ready Minutes  \
Rep ID Branch                                                          
Alice  South           0          0           199                 46   
Amanda North           0          1           229                 43   
Andy   North           0          3           228                 43   
Brent  North           0          1           362                 71   
Cam    North           1          0           233                 46   
Duke   North           2          0           292                 56   
Eric   South           0          0           361                 54   
George South           0          0           298                 53   
Helga  South           0          2           292                 58   
Joe    North           0          0           256                 49   
Josh   South           3          0           260                 50   
Karl   South           1          0           287               

#### Manipulation Question 4: For each branch and the entire company, what is the average speed of answer?
##### Record your answer by double clicking here:
Average speed of answer (rounded to nearest hundredth):
- North Branch in seconds = 
- South Branch in seconds = 
- Company Wide in seconds = 


In [ ]:
# Code Used 


<span style="background-color:orangered; color:orangered;">- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -</span>

## Visualization

Create a visualization for each of the following questions. Some of the code to handle aggregating and storing data may be written for you. For each visualization, you choose the chart style that you feel suits the situation best. Make note of the chart style you chose and why. 

*NOTE Some questions you may decide to use more than one chart and or chart style.

#### Visualization 1:  What is the average abandonment rate per queue?
##### Record your answer by double clicking here:

> Chart style you chose and why: 

In [ ]:
# Create visualization 1 here
# Import any additional libraries needed with alias



# The dictionary abandonment_rates has the data you need.
abandonment_rates = {}
queues = ["A", "B", "C", "D"]
queue_dict = df.groupby("Queue").agg("sum")
for i in range(4):
    abandonment_rates[queues[i]] = queue_dict["Abandoned"][i] / queue_dict["Calls"][i] 

#Your Code Here:




#### Visualization 2: What is the service level percentage and average speed of answer for each rep in the North Branch?
##### Record your answer by double clicking here:

> Chart style you chose and why: 

In [ ]:
# Create visualization 2 here
# north_plt contains the data you need for the average speed of answer of each rep

north_plt = north.groupby("Rep ID")["Incoming Wait Time"].mean().to_frame().reset_index()

# Finding each Rep's Personal Service Level Percentage.  Basically, Calls within 2 secs / total calls

# Table 1: Total Incoming calls less than 2 seconds grouped by Rep
quick_calls = north[north["Incoming Wait Time"] <= 2.0]
quick_reps = quick_calls[["Rep ID", "Calls"]]
quick_stats = quick_reps.groupby(["Rep ID"]).sum()  # Final Table


# Table 2: Total Incoming Calls Only grouped by Rep
total_calls_in = north[north["Incoming or Outgoing"] == "Incoming"]
rep_calls = total_calls_in[["Rep ID", "Calls"]]     
total_stats = rep_calls.groupby(["Rep ID"]).sum() # Final Table  

#  Table 3: Service Level Percentage created via merge
service_level = pd.merge(quick_stats, total_stats, on="Rep ID")

# Create Percentage Column in Table 3
service_level["Service Level %"] = service_level["Calls_x"]/service_level["Calls_y"] * 100

#Your Code Here:


#### Visualization 3: For each type of call purpose, how many calls are outgoing vs. incoming?
##### Record your answer by double clicking here:

> Chart style you chose and why: 

In [ ]:
# Create visualization 3 here:
# The three dictionaries, complaints, sales_support, and product_support, have the information you need

purpose_group = df.groupby("Call Purpose")
call_purpose = ["Complaint", "Product Support", "Sales Support"]
purpose_counts = purpose_group["Incoming or Outgoing"].value_counts()
print(purpose_counts)

complaints = purpose_counts["Complaint"].to_dict()
sales_support = purpose_counts["Sales Support"].to_dict()
product_support = purpose_counts["Product Support"].to_dict()

#Your Code Here:


<span style="background-color:orangered; color:orangered;">- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -</span>

## Summarize Your Work

With what you know now about the two call centers and the entire company, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.


#### Summary Question 1:  Using KPIs such as average abandonment rate, service level and average speed of answer, in your opinion, which one of the two branches is operating more efficiently? Why?

##### Record your answer by double clicking here:
> Your Answer: 



#### Summary Question 2: Based on the number of reps in each branch and how quickly the reps are working, in your opinion, which branch would benefit from the extra help?

##### Record your answer by double clicking here:
> Your Answer: 


#### Summary Question 3: Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?

##### Record your answer by double clicking here:
> Your Answer: 


<span style="background-color:orangered; color:orangered;">- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -</span>

## Bonus Mission
Create a visualization that answers this question: For each call purpose, how many calls (incoming and outgoing) take place in each time block?
##### Record your answer by double clicking here:

> Chart style you chose and why: 

In [ ]:
# Create your Bonus Mission visualization here!
call_times = df[["Time Block", "Call Purpose", "Incoming or Outgoing", "Calls"]]

# Use groupby to plot based on time blocks:

# Use groupby and get_group to select which call purpose to plot: